localstack/aws/protocol/parser.py

```diff
@@ -1063,17 +1062,21 @@ def _parse_shape(
         Special handling of parsing the shape for s3 object-names (=key):
         trailing '/' are valid and need to be preserved, however, the url-matcher removes it from the key
         we check the request.url to verify the name.
-        We decode the key for the comparison with `base_url` in case of special characters.
+        Trailing slashes are stripped from the path, so we need special logic to compare the parsed Key parameter
+        against the path and add the missing slashes
         """
         if (
             shape is not None
             and uri_params is not None
             and shape.serialization.get("location") == "uri"
             and shape.serialization.get("name") == "Key"
-            and request.base_url.endswith(f"{unquote(uri_params['Key'])}/")
+            and (
+                (trailing_slashes := request.path.partition(uri_params["Key"])[2])
+                and all(char == "/" for char in trailing_slashes)
+            )
```

Comment:
If I see this correctly, this now takes everything after the key, and then checks if there are only slashes afterwards. Afterwards we add (the arbitrary amount of slashes) back to the parsed key. Is that correct?
So you solution now works for 0 to arbitrary trailing slashes, is that correct?
How exactly does this fix issues with spaces in the key name?

Reply:
The issue with space was the call to `unquote` of the key, which was compared a "quoted" path. So this was one part of the fix, and I've also added the logic to handle multiple trailing slashes.

Reply:
Ah, right, I missed the removal of unquote in the changeset. That makes sense, thanks!